Firebase/Database/Persistence/FLevelDBStorageEngine.m

```diff
@@ -132,6 +132,9 @@ - (void)runMigration {
         }
     } else if ([oldVersion isEqualToString:kFPersistenceVersion]) {
         // Everythings fine no need for migration
+    } else if ([oldVersion length] == 0) {
```

Comment:
Should we instead eliminate `runMigration` completely, get rid of the file management, and add it back in the unlikely case that there is ever a version 2?

Reply:
I think the code is meant to crash for downgrades - e.g. when we add version 2, version 1 clients are supposed to crash.

Reply:
Understood, but isn't it unlikely we'll ever add a version 2? In which case it would be nice to stop managing something in the filesystem.  We could always turn this code on again if and when version 2 happens.

I'm ok with whatever you decide.

Reply:
I do think it is unlikely, but I prefer that we leave this check since we cannot turn this check back on in code that we already shipped. Without this check, it would be a much more expensive to add schema v2, as it that would require us to ship an SDK with version 1 that rejected version 2 schema before even considering a migration.